homework/shared_ptr/WeakPtr-ut.cpp

```diff
@@ -68,3 +69,28 @@ TEST(WeakPtrTest, WeakPtrMoveAssignment) {
     ASSERT_FALSE(weakPtr2.expired());
 }
 
+TEST(WeakPtrTest, WeakPtrCopyAssignment) {
+    cs::SharedPtr<int> sharedPtr{new int{WeakPtrTestConst::intValue}};
+    cs::WeakPtr<int> weakPtr{sharedPtr};
+    cs::WeakPtr<int> weakPtr2 = weakPtr;
```

Comment:
copy constructor is used here. Use this form
```suggestion
    cs::WeakPtr<int> weakPtr2{some_other_shared_ptr};
    weakPtr2 = weakPtr;
```